### PR TITLE
Enable utf8 encoding

### PR DESCRIPTION
**Pull request by it-hms**
- Additional constructor with boolean parameter enable utf8 parameter.
 - Additional method to publish MQTT message w/ config utf8.
 - Increase version to 1.2.

### CODE REVIEW

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 1]
```diff
@@ -42,6 +52,41 @@ public MqttManager(String mqttId, String mqttHost) throws Exception {
     setProtocolVersion(MqttConstants.MQTT_PROTOCOL_VERSION_OPTION_DEFAULT);
   }
 
+  /**
+   * Additional constructor for MqttManager class. Required parameters MQTT Client ID, MQTT Broker
+   * URL and boolean for Enable UTF8 conversion of payload. UTF8 is recommended when tag names or
+   * tag values may have non-ascii values.
+   *
+   * @param mqttId MQTT client ID
+   * @param mqttHost MQTT broker URL
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java:61]: missing @ param for enableUtf8 boolean

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 2]
```diff
@@ -25,6 +26,15 @@
...
   private boolean mqttThreadRun = true;
 
+  /**
+   * Boolean flag to enable UTF8 string conversion. Will enable non-ascii characters, but decreases
+   * performance by a significant amount. Defaults to false, and can be set by constructor.
+   */
+  private boolean mqttUtf8Convert = false;
+
+  /** Encoding format for strings */
+  private final String UTF_8 = "UTF-8";
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
  private static final String UTF_8 = "UTF-8";
```
This variable should be a 'static final' constant. The ALL_CAPS naming convention only applies to 'static final' constants as well.

Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
  private static final String UTF_8 = "UTF-8";
```

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 3]
```diff
@@ -42,6 +52,41 @@ public MqttManager(String mqttId, String mqttHost) throws Exception {
     setProtocolVersion(MqttConstants.MQTT_PROTOCOL_VERSION_OPTION_DEFAULT);
   }
 
+  /**
+   * Additional constructor for MqttManager class. Required parameters MQTT Client ID, MQTT Broker
+   * URL and boolean for Enable UTF8 conversion of payload. UTF8 is recommended when tag names or
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
   * URL and boolean for Enable UTF-8 conversion of payload. UTF-8 is recommended when tag names or
```

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 4]
```diff
@@ -25,6 +26,15 @@
   /** Boolean flag to control the MQTT thread loop. */
   private boolean mqttThreadRun = true;
 
+  /**
+   * Boolean flag to enable UTF8 string conversion. Will enable non-ascii characters, but decreases
```
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
   * Boolean flag to enable UTF-8 string conversion. Will enable non-ASCII characters, but decreases
``

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 5]
```diff
@@ -25,6 +26,15 @@
...
+   * performance by a significant amount. Defaults to false, and can be set by constructor.
+   */
+  private boolean mqttUtf8Convert = false;
+
+  /** Encoding format for strings */
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
  /** UTF-8 charset name */
```

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java [hunk 6]
```diff
@@ -42,6 +52,41 @@ public MqttManager(String mqttId, String mqttHost) throws Exception {
...
+   * tag values may have non-ascii values.
+   *
+   * @param mqttId MQTT client ID
+   * @param mqttHost MQTT broker URL
+   * @throws Exception if unable to create MQTT client
+   */
+  public MqttManager(String mqttId, String mqttHost, boolean enableUtf8) throws Exception {
+    this(mqttId, mqttHost);
+    this.mqttUtf8Convert = enableUtf8;
+  }
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java:68]:
```suggestion
  }

```

File: src/com/hms_networks/americas/sc/mqtt/package.html [hunk 7]
```diff
@@ -2,7 +2,7 @@
 <BODY>
 Ewon Flexy MQTT library created by the HMS Networks MU Americas Solution Center.
 
-@version 1.1
+@version 1.2
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/mqtt/package.html:5]: This should be a separate commit titled something like "Rev'd version number to v1.2"

File: src/com/hms_networks/americas/sc/mqtt/MqttManager.java (hunk 2 above)
Comment (alexjhawk) [inline src/com/hms_networks/americas/sc/mqtt/MqttManager.java]:
```suggestion
  /** Encoding format for strings */
  private final String STRING_ENCODING_FORMAT = "UTF-8";
```
```suggestion
  /** UTF-8 encoding format for strings */
  private final String UTF_8 = "UTF-8";
```
This might be more of a personal preference, but I feel like the current Javadoc comment doesn't entirely line up with what the variable name is, and one of the above would more clearly align the comment and variable name.

**Review by TomKimsey — CHANGES_REQUESTED** [review]

**Review by alexjhawk — CHANGES_REQUESTED** [review]

**Review by TomKimsey — APPROVED** [review]

**Review by alexjhawk — APPROVED** [review]
